packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -149,6 +151,14 @@ async function startWatcher(opts: SetupOpts) {
 
   const distDir = path.join(opts.dir, opts.nextConfig.distDir)
 
+  // we ensure the types directory exists on turbo
+  if (usingTypeScript && opts.turbo) {
+    const distTypesDir = path.join(distDir, 'types')
+    if (!fs.existsSync(distTypesDir)) {
+      await mkdir(distTypesDir, { recursive: true })
+    }
+  }
```

Comment:
@huozhi Since turbo don't generate `types` dir like our webpack plugin does, we ensure it is generated here for further `typed...` features.

Reply:
Empty `env.d.ts` don't affect existing envs:

https://github.com/vercel/next.js/assets/120007119/1d9b8be0-aed6-4efd-8610-5a7afa4e45b8

Reply:
Can we check if types dir is created in the type creation util so we don't have to do that logic for turbopack specifically? This might be easily forgotten to update later on